File: samples/spring-mvc/src/main/java/io/openapiprocessor/samples/BarController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.BarApi;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class BarController implements BarApi {

    @Override
    public MappedBar postBar(MappedBar bar) {
        return bar != null ? bar : new MappedBar();
    }
}

File: samples/spring-mvc-boot-3-monolith-kt/src/main/kotlin/io/openapiprocessor/samples/bar/api/BarController.kt
package io.openapiprocessor.samples.bar.api

import io.openapiprocessor.samples.bar.domain.BarService
import io.openapiprocessor.samples.bar.model.BarDto
import org.springframework.web.bind.annotation.RestController

@RestController
class BarController(private val mapper: BarMapper, private val service: BarService): BarApi {

    override fun postBar(bar: BarDto): BarDto {
        return mapper.map(service.echo(mapper.map(bar)))
    }
}
